collect_app/src/main/java/org/odk/collect/android/activities/BearingActivity.java

```diff
@@ -69,44 +68,36 @@ protected void onResume() {
         super.onResume();
         sensorManager.registerListener(this, accelerometer, SensorManager.SENSOR_DELAY_GAME);
         sensorManager.registerListener(this, magnetometer, SensorManager.SENSOR_DELAY_GAME);
-        bearingDialog.show();
+        bearingDialog.show(getSupportFragmentManager(), ProgressDialogFragment.COLLECT_PROGRESS_DIALOG_TAG);
     }
 
     /**
      * Sets up the look and actions for the progress dialog while the compass is
      * searching.
      */
     private void setupBearingDialog() {
-        // dialog displayed while fetching bearing
-        bearingDialog = new DayNightProgressDialog(this);
         DialogInterface.OnClickListener geopointButtonListener =
-                new DialogInterface.OnClickListener() {
-                    @Override
-                    public void onClick(DialogInterface dialog, int which) {
-                        switch (which) {
-                            case DialogInterface.BUTTON_POSITIVE:
-                                returnBearing();
-                                break;
-                            case DialogInterface.BUTTON_NEGATIVE:
-                                bearingDecimal = null;
-                                finish();
-                                break;
-                        }
+                (dialog, which) -> {
+                    switch (which) {
+                        case DialogInterface.BUTTON_POSITIVE:
+                            returnBearing();
+                            break;
+                        case DialogInterface.BUTTON_NEGATIVE:
+                            bearingDecimal = null;
+                            finish();
+                            break;
                     }
                 };
 
-        // back button doesn't cancel
-        bearingDialog.setCancelable(false);
-        bearingDialog.setIndeterminate(true);
-        bearingDialog.setIcon(android.R.drawable.ic_dialog_info);
-        bearingDialog.setTitle(getString(R.string.getting_bearing));
-        bearingDialog.setMessage(getString(R.string.please_wait_long));
-        bearingDialog.setButton(DialogInterface.BUTTON_POSITIVE,
-                getString(R.string.accept_bearing),
-                geopointButtonListener);
-        bearingDialog.setButton(DialogInterface.BUTTON_NEGATIVE,
-                getString(R.string.cancel_location),
-                geopointButtonListener);
+        ProgressDialogFragment progressDialogFragment = new ProgressDialogFragment();
```

Comment:
I think it would be better to have a new implementation of ProgressDialogFragment for this case. Please take a look at what we did recently for ResetProgressDialog: https://github.com/getodk/collect/pull/4980/files

Reply:
And other dialogs should also have their own implementation do you agree @seadowg?